numpy/linalg/tests/test_linalg.py

```diff
@@ -1507,6 +1507,30 @@ def test_0_size(self):
 class TestCholesky(object):
     # TODO: are there no other tests for cholesky?
 
+    def test_basic_property(self):
+        # Check A = L L^H
+        shapes = [(1, 1), (2, 2), (3, 3), (50, 50), (3, 10, 10)]
+        dtypes = (np.float32, np.float64, np.complex64, np.complex128)
+
+        for shape, dtype in itertools.product(shapes, dtypes):
+            np.random.seed(1)
+            a = np.random.randn(*shape)
+            if np.issubdtype(dtype, np.complexfloating):
+                a = a + 1j*np.random.randn(*shape)
+
+            t = list(range(len(shape)))
+            t[-2:] = -1, -2
+
+            a = np.matmul(a.transpose(t).conj(), a)
+            a = np.asarray(a, dtype=dtype)
```

Comment:
Out of curiousity, is the `dtype` actually needed here?

Reply:
Of course, numpy.linalg doesn't do anything apart from double precision, but better still to test all combinations.